src/grub2/grub-static-greenboot.cfg

```diff
@@ -0,0 +1,19 @@
+# greenboot support, aka boot counter and boot success reporting
+insmod increment
+# Check if boot_counter exists and boot_success=0 to activate this behaviour.
+if [ -n "${boot_counter}" -a "${boot_success}" = "0" ]; then
+  # if countdown has ended, choose to boot rollback deployment,
+  # i.e. default=1 on OSTree-based systems.
+  if  [ "${boot_counter}" = "0" -o "${boot_counter}" = "-1" ]; then
```

Comment:
why forcing 0 to be -1 when it can happen with `decrement boot_counter` automatically , I have seen similar things [here](https://github.com/osbuild/osbuild/blob/3cc733d7cd3d2e65d76e227d56ef8a47ce32f469/stages/org.osbuild.grub2#L299) , couldn't understand the logic behind this.

Reply:
I have no idea either, I'm just copying it from the osbuild code.

But actually the more I think about this, the more it feels like the right move actually is to keep these grub drop-ins in `ignition.rpm` and `greenboot.rpm` respectively in say `/usr/lib/bootupd/grub2-static` or so.  WDYT?

Reply:
having it in rpm makes sense, though I am not sure how to set that up in [grub.cfg](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system_administrators_guide/ch-working_with_the_grub_2_boot_loader#doc-wrapper) , which image builder writes it directly in the [config](https://github.com/osbuild/osbuild/blob/3cc733d7cd3d2e65d76e227d56ef8a47ce32f469/stages/org.osbuild.grub2#L591C18-L591C22)

Reply:
Seems fair.
May not be a appropriate question, still I am going to ask to clear my doubts: Though I dont see many user doing this , but  using rpm installer  will require additional  `bootupctl backend install` to be run again to update the bootloader configs?

Reply:
We don't directly support updating the configs yet.

(In practice of course, it will trivially work to just `cp` them manually)